tests/tests_switch_provider.yml

```diff
@@ -6,5 +6,8 @@
 - import_playbook: playbooks/tests_switch_provider.yml
   when:
     # The test requires or should run with NetworkManager, therefore it cannot
-    # run on RHEL/CentOS 6
-    - ansible_distribution_major_version != '6'
+    # run on RHEL/CentOS 6. In addition it cannot run on CentOS/RedHat 9.
+    - (ansible_distribution in ['centOS', 'RedHat'] and
+      ansible_distribution_major_version != '6' and
+      ansible_distribution_major_version | int < 9) or
```

Comment:
IMHO these two when conditions are enouhg (!= 6 and < 9). Restricting to the distribution adds unnecessary complexity IMHO.

Reply:
I think you are right. This is fine to me :-)